lib/puppet/http/client.rb

```diff
@@ -140,14 +151,19 @@ def close
 
   private
 
-  def execute_streaming(request, user: nil, password: nil, ssl_context:, include_system_store:, &block)
+  def execute_streaming(request, options, &block)
+    user = options.fetch(:user, nil)
+    password = options.fetch(:password, nil)
+    ssl_context = options.fetch(:ssl_context, nil)
+    include_system_store = options.fetch(:include_system_store, false)
+
     redirects = 0
     retries = 0
     response = nil
     done = false
 
     while !done do
-      connect(request.uri, ssl_context: ssl_context, include_system_store: include_system_store) do |http|
+      connect(request.uri, options: {ssl_context: ssl_context, include_system_store: include_system_store}) do |http|
```

Comment:
same here. Rather than construct a new options hash, I thought I should be able to just pass through the options hash that was passed into `execute_streaming`. That caused a bunch of errors though, so I'm still digging into that.